hera_sim/visibilities/vis_cpu.py

```diff
@@ -316,58 +279,71 @@ def _base_simulate(self, crd_eq, I_sky):
             myid = self.mpi_comm.Get_rank()
             nproc = self.mpi_comm.Get_size()
 
+        if self.correct_source_positions:
+            # TODO: check if this is the right time to be using...
+            ra, dec = self.correct_point_source_pos(
+                data_model, obstime=Time(data_model.uvdata.time_array[0], format="jd")
+            )
+        else:
+            ra, dec = data_model.sky_model.ra, data_model.sky_model.dec
+
+        crd_eq = convs.point_source_crd_eq(ra, dec)
+
         # Convert equatorial to topocentric coords
-        eq2tops = self.get_eq2tops()
+        eq2tops = self.get_eq2tops(data_model.uvdata, data_model.lsts)
+
+        # ant_list = data_model.uvdata.get_ants()
+        # The following are antenna positions in the order that they are
+        # in the uvdata.data_array
+        active_antpos, ant_list = data_model.uvdata.get_ENU_antpos(pick_data_ants=True)
 
         # Get pixelized beams if required
         if self.use_pixel_beams:
-            beam_lm = self.get_beam_lm()
-            # polarized=True:  (NANT, NAXES, NFEEDS, NFREQS, BM_PIX, BM_PIX)
-            # polarized=False: (NANT, NFREQS, BM_PIX, BM_PIX)
-            # if not self.polarized:
-            #    beam_lm = beam_lm[np.newaxis, np.newaxis, :, :, :]
+            beam_lm = self.get_beam_lm(data_model)
+            if self.polarized:
+                beam_lm = np.transpose(beam_lm, (1, 2, 0, 3, 4, 5))
         else:
             beam_list = [
-                self.beams[np.where(self.beam_ids == ant)[0][0]]
-                for ant in self.ant_list
+                data_model.beams[data_model.beam_ids[name]]
+                for number, name in zip(
+                    data_model.uvdata.antenna_numbers, data_model.uvdata.antenna_names
+                )
+                if number in ant_list
             ]
 
+        # Get x_orientation
+        x_orient = data_model.uvdata.x_orientation
+        if x_orient is None:
+            data_model.uvdata.x_orientation = "e"  # set in UVData object
+            x_orient = "e"  # default to east
```

Comment:
`x_orientation` has to be either `"east"` or `"north"` (full word). See bottom of https://pyuvdata.readthedocs.io/en/latest/uvdata_parameters.html